strongbox-web-core/src/main/java/org/carlspring/strongbox/web/HeaderMappingFilter.java

```diff
@@ -55,14 +62,27 @@
     private static final String USER_AGENT_MAVEN = "Maven";
 
     private Map<String, String> userAgentMap = new HashMap<>();
+    private Map<String, String> layoutMap = new HashMap<>();
+
+    @Inject
+    private ConfigurationManager configurationManager;
 
     @Override
     public void init(FilterConfig filterConfig)
         throws ServletException
     {
+        //Do nothing
+    }
+
+    @PostConstruct
+    public void postConstruct()
```

Comment:
It should be possible to populate this via some sort of registry, or something. Maybe the respective provider could have a `getUserAgent()`, or something... If the `LayoutProviderRegistry` is injected here, you could loop over the registered providers and get it a bit more dynamically, because at some point, hopefully, people might implement their own providers that are not part of the standard distribution and it will be hard to register them without such a dynamic approach.